Petits-debuts-api/server.js

```diff
@@ -324,8 +335,71 @@ app.post("/money/update", async (req, res) => {
 });
 //end
 
-//end
+//end of routes
+//node-cron to run at midnight every day
+cron.schedule("0 0 * * *", function () {
+  const isServiceOrder = (obj) => {
+    const firstValue = obj[Object.keys(obj)[0]];
+    if (typeof firstValue !== "number") {
+      return true;
+    } else {
+      return false;
+    }
 
+    //return firstValue instanceof Date;
+  };
+  const getAllServiceOrders = async () => {
+    try {
+      const serviceOrder = await Order.findAll({
+        include: [{ model: User, as: "user" }],
+        order: [["createdAt", "DESC"]],
+      });
+      for (let i = 0; i < serviceOrder.length; i++) {
+        console.log("inside for loop");
+        if (isServiceOrder(serviceOrder[i].order)) {
+          console.log("inside if statement");
+          for (const [key, value] of Object.entries(serviceOrder[i].order)) {
+            const today = new Date();
+            const appointment = new Date(value);
+            const day = appointment.getUTCDate().toString().padStart(2, "0");
+            const month = (appointment.getUTCMonth() + 1)
+              .toString()
+              .padStart(2, "0");
+
+            if (appointment > today) {
+              const product = await Product.findOne({
+                where: { id: key },
+              });
+              const msg = {
+                to: "chinyereoffor@meta.com", // Change to your recipient
```

Comment:
This is used for testing only, right? The recipient can change based on the order, and should be from the Service Order

Reply:
yes, but for now, i used my meta email just so we can demonstrate that it works
I could change it to the appropriate value after presentation.